agent/replacecall/src/main/java/randoop/instrument/ReplaceCallAgent.java

```diff
@@ -0,0 +1,201 @@
+ *
+ * <p>The classes of packages listed in the resource file {@code "default-load-exclusions.txt"} are
+ * excluded from transformation.
+ */
+public class ReplaceCallAgent {
+
+  /** Run the replacecall agent in debug mode. */
+  @SuppressWarnings("WeakerAccess")
+  @Option("print debug information")
+  public static boolean debug = false;
+
+  @SuppressWarnings("WeakerAccess")
+  @Option("directory name where debug logs are written")
+  public static String debug_directory = null;
+
+  static Path debugPath;
+
+  @SuppressWarnings("WeakerAccess")
+  @Option("print progress information")
+  public static boolean verbose = false;
+
+  /** The file from which to read the user replacements for replacing calls. */
+  @SuppressWarnings("WeakerAccess")
+  @Option("file listing methods whose calls to replace by substitute methods")
+  public static File map_calls = null;
+
+  /** Exclude transformation of classes in the the listed packages. */
+  @SuppressWarnings("WeakerAccess")
+  @Option("file listing packages whose classes should not be transformed")
+  public static File dont_transform = null;
+
+  /**
+   * Entry point of the replacecall Java agent. Initializes the {@link CallReplacementTransformer}
+   * so that when classes are loaded they are transformed to replace calls to methods as specified
+   * in the replacements file(s).
+   *
+   * @param agentArgs the arguments to the agent
+   * @param inst the {@code Instrumentation} object
+   * @throws IOException if there is an error reading a file
+   */
+  public static void premain(String agentArgs, Instrumentation inst) throws IOException {
+    if (verbose) {
+      System.out.format(
+          "In premain, agentargs ='%s', " + "Instrumentation = '%s'%n", agentArgs, inst);
+    }
+
+    if (agentArgs != null) { // If there are any arguments, parse them
+      Options options = new Options(ReplaceCallAgent.class);
+      String[] target_args = options.parse_or_usage(agentArgs);
+      if (target_args.length > 0) {
+        System.err.printf("Unexpected agent arguments %s%n", Arrays.toString(target_args));
+        System.exit(1); // Exit on bad user input.
+      }
+    }
+
+    debugPath = Paths.get("").toAbsolutePath().normalize();
+    if (debug && debug_directory != null && !debug_directory.isEmpty()) {
+      debugPath = debugPath.resolve(debug_directory);
+      if (!Files.exists(debugPath)) {
+        Files.createDirectory(debugPath);
+      }
+    }
+
+    // Load package prefixes from the resource file in the jar for default package exclusions
+    Set<String> excludedPackagePrefixes = new LinkedHashSet<>();
+
+    String exclusionFileName = "/default-load-exclusions.txt";
+    InputStream inputStream = ReplaceCallAgent.class.getResourceAsStream(exclusionFileName);
+    if (inputStream == null) {
+      System.err.println("Unable to find default package exclusion file. Please report.");
+      System.exit(1); // Exit on configuration error. (Throwing exception would halt JVM.)
```

Comment:
I wrapped most of premain with `try-catch` for `Throwable` that prints a message and calls `System.exit`. If an error occurs that is due to configuration error, the private exception `BugInAgentException` is thrown. 

(`BugInRandoopException` is a class within Randoop and is not available to the agent because the agent does not depend on Randoop.)